### PR TITLE
improvements to webform 5 and signature Lac Farmer Profile

### PR DESCRIPTION
**Pull request by jgomezr**
improvements to webform 5 to show client list and make changes on LacFarmerDetailPageController to show the signature on Lac Farmer Details profile

### CODE REVIEW

**Comment by lc0rp** [comment]
I'm going to merge this, but it's unclear what you were trying to do. Next time, please make the title a bit more descriptive. e.g. "Improvements to webform 5 to add a field that shows the client name".

File: src/classes/AlsurOrderDelivery.cls [hunk 1]
```diff
@@ -17,7 +22,9 @@ public PageReference executeSearch() {
                 Order_Code__c,
                 Number_of_Baskets__c
           from CPA_Order_Allocation__c
-          where Id = :nameQuery];       
+          where Client__c = :nameQuery
+          and Association__r.Name =:asociation
+          and Id != :deliveri];
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: "deliveri" is a Set. Where you trying to say the following?

```
and Id not in :deliveri
```

File: src/classes/AlsurOrderDelivery.cls [hunk 2]
```diff
@@ -53,16 +60,22 @@ public List<SelectOption> getclientItems() {
                                                   from Association_CPA_Order_Delivery__c]){ 
     del.add(deli.CPA_Order_Allocation__c);
   }
+  Set<String> cli = new Set<String>();
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: Please use full names for your variables. clients, not cli

File: src/classes/AlsurOrderDelivery.cls [hunk 3]
```diff
@@ -53,16 +60,22 @@ public List<SelectOption> getclientItems() {
...
+  for (CPA_Order_Allocation__c clie : [select Client__c
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: client not clie

File: src/classes/AlsurOrderDelivery.cls [hunk 4]
```diff
@@ -5,6 +5,11 @@ public List<CPA_Order_Allocation__c> orders {get; set;}
 public Association_CPA_Order_Delivery__c allocation { get; set; }
 
 public PageReference executeSearch() {
+  Set<ID> deliveri = new Set<ID>();
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: deliveries not deliveri

File: src/classes/AlsurOrderDelivery.cls [hunk 5]
```diff
@@ -5,6 +5,11 @@ public List<CPA_Order_Allocation__c> orders {get; set;}
...
+  for (Association_CPA_Order_Delivery__c deliveries : [select CPA_Order_Allocation__c
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls:9]: I think this is singular, so it should be delivery not deliveries

File: src/classes/AlsurOrderDelivery.cls [hunk 6]
```diff
@@ -17,7 +22,9 @@ public PageReference executeSearch() {
                 Order_Code__c,
                 Number_of_Baskets__c
           from CPA_Order_Allocation__c
-          where Id = :nameQuery];       
+          where Client__c = :nameQuery
+          and Association__r.Name =:association
+          and Id != :delivery];
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: This hasn't change yet. Delivery looks like a collection to me. I don't think you can say

```
Id != : Set;
```

I think you want to say

```
Id not in: Set;
```

File: src/classes/AlsurOrderDelivery.cls [hunk 7]
```diff
@@ -26,43 +33,49 @@ public AlsurOrderDelivery(){
...
+    delivery1.add(deliveries1.CPA_Order_Allocation__c);
   }
-  Set<ID> aso= new Set<ID>();
-  for (CPA_Order_Allocation__c asos : [select Association__c 
+  Set<ID> association1= new Set<ID>();
+  for (CPA_Order_Allocation__c associations1 : [select Association__c 
                                         from CPA_Order_Allocation__c 
-                                        where Id != :del1]){ 
-    aso.add(asos.Association__c);
+                                        where Id != :delivery1]){
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: Same thing here, delivery1 is a collection. I think you want to say 

```
Id not in :delivery;
```

File: src/classes/AlsurOrderDelivery.cls [hunk 8]
```diff
@@ -26,43 +33,49 @@ public AlsurOrderDelivery(){
...
                                         from CPA_Order_Allocation__c 
-                                        where Id != :del1]){ 
-    aso.add(asos.Association__c);
+                                        where Id != :delivery1]){ 
+    association1.add(associations1.Association__c);
   }
   list<Association__c> associations = [Select Name 
                                         from Association__c 
-                                        where Id = :aso];
+                                        where Id = :association1];
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: Same here.

File: src/classes/AlsurOrderDelivery.cls [hunk 9]
```diff
@@ -26,43 +33,49 @@ public AlsurOrderDelivery(){
...
+  Set<ID> delivery2 = new Set<ID>();
+  for (Association_CPA_Order_Delivery__c deliveries2 : [select CPA_Order_Allocation__c 
                                                   from Association_CPA_Order_Delivery__c]){ 
-    del.add(deli.CPA_Order_Allocation__c);
+    delivery2.add(deliveries2.CPA_Order_Allocation__c);
+  }
+  Set<String> client = new Set<String>();
+  for (CPA_Order_Allocation__c clients2 : [select Client__c
+                                        from CPA_Order_Allocation__c 
+                                        where Id != :delivery2]){
```
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: Same here. Do you have tests for this code? I think if you write some positive and negative tests, you'd see whether this actually works. Let's discuss.

**Comment by jgomezr** [comment]
ok luke changes done

File: src/classes/AlsurOrderDelivery.cls (hunk 8 above)
Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: This is still missing. I'd like to see some tests too.

File: src/classes/LacFarmerProfilePageController.cls [hunk 10]
```diff
@@ -133,4 +155,309 @@ public with sharing class LacFarmerProfilePageController {
...
+            
+       PMF_Finca_Categoria_2__c millCat = [Select Nivel_categor_a__c, Tipo__c, Fecha__c from PMF_Finca_Categoria_2__c where Plan_de_Manejo_de_Finca__c = :fmp.Id];
+       FMPCategoryResult res2 = new FMPCategoryResult();
+      res2.setLevel(millCat.Nivel_categor_a__c);
+      res2.setResultDate(millCat.Fecha__c);
+      res2.setResultType(millCat.Tipo__c);
+        
+      resultsGroup.addResultGroup(res1);
+      resultsGroup.addResultGroup(res2);
+      /*
```
Comment (lc0rp) [inline src/classes/LacFarmerProfilePageController.cls:279]: This code is commented out. Do we need it? If not, let's get rid of it.

It'd also be good to add some comments to the code.

File: src/classes/LacFarmerProfilePageController.cls [hunk 11]
```diff
@@ -133,4 +155,309 @@ public with sharing class LacFarmerProfilePageController {
...
+       PMF_Finca_Categoria_8_Gestion_ambienta__c envMgmtCat = [Select Nivel_categoria__c from PMF_Finca_Categoria_8_Gestion_ambienta__c where Plan_de_Manejo_de_Finca__c = :fmp.Id];
+       PMF_Finca_Categ_9_Cosecha_PostCosecha__c harvestPostHarvestCat = [Select Nivel_categoria__c from PMF_Finca_Categ_9_Cosecha_PostCosecha__c where Plan_de_Manejo_de_Finca__c = :fmp.Id];
+       PMF_Finca_Categoria_10_Lotes__c lotsCat = [Select Nivel_categoria__c from PMF_Finca_Categoria_10_Lotes__c where Plan_de_Manejo_de_Finca__c = :fmp.Id];
+       PMF_Finca_Categoria_11_Alm_cigo__c seedbedCat = [Select Nivel_categoria__c from PMF_Finca_Categoria_11_Alm_cigo__c where Plan_de_Manejo_de_Finca__c = :fmp.Id];
+       */
+       results.add(resultsGroup);
+     } 
+    return results;   
+    }
+    /*
```
Comment (lc0rp) [inline src/classes/LacFarmerProfilePageController.cls:294]: Remove this code if we don't need it.

**Comment by jgomezr** [comment]
luke with LacFarmerProfilePageController.cls I did was pass as it is in the production instance

File: src/classes/AlsurOrderDelivery.cls (hunk 8 above)
Comment (jgomezr) [inline src/classes/AlsurOrderDelivery.cls]: These lines run smoothly, and the probe in my instance and have operated normally

Comment (lc0rp) [inline src/classes/AlsurOrderDelivery.cls]: The syntax seems wrong to me. If association1 is a collection, then you don't use "Id =", you say "Id in" - so either I'm reading the code wrong, or there's a new apex feature I don't know about. It'd be great to see a test that shows that it actually returns the desired results.

Comment (jgomezr) [inline src/classes/AlsurOrderDelivery.cls]: ok, make change from  "=" to "in"

**Pull request merged**